Bio/Seq.py

```diff
@@ -2052,6 +2052,22 @@ def ungap(self, gap="-"):
             raise ValueError(f"Unexpected gap character, {gap!r}")
         return self.replace(gap, b"")
 
+    @property
+    def defined(self):
+        """Return true if the sequence is defined, false if undefined or partially defined.
+
+        Zero-length sequences are always considered to be defined.
+        """
+        if len(self) == 0:
+            return True
```

Comment:
This is OK for now, but we can remove these two lines once we disallow zero-length undefined sequences.